docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,238 @@
+
+    function eq(Coin a, Coin b) pure returns (bool) {
+        return Coin.unwrap(a) == Coin.unwrap(b);
+    }
+
+    function c(uint mantissa, uint exponent) pure suffix returns (Coin) {
+        return Coin.wrap(mantissa * 10**6 / 10**exponent);
+    }
+
+    function uc(uint microValue) pure suffix returns (Coin) {
+        return Coin.wrap(microValue);
+    }
+
+    contract CoinSeller {
+        mapping(address => Coin) balances;
+
+        function buy() public payable {
+            require(msg.value == 1 ether);
+            assert(1.5 c == 1_500_000 uc);
+            balances[msg.sender] = balances[msg.sender] + 1.5 c;
+        }
+    }
+
+.. index:: ! literal suffix; definition
+
+Defining Suffix Functions
+-------------------------
+
+Literal suffixes can be defined by applying the built-in ``suffix`` modifier to a free function.
+
+Only pure functions can be used as suffixes.
+This means that suffixes cannot read or modify contract state.
+As all pure functions, however, they can perform pure external calls.
+
+.. index:: literal;address
+
+Suffix Function Parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A suffix function must accept and return exactly one value.
+As a special case, suffixes on :ref:`rational literals<rational_literals>` can optionally accept two arguments,
+produced by the :ref:`fractional decomposition<fractional_decomposition>` of such a literal.
+
+Suffixes can only have parameters of types for which an implicit conversion from a literal exists.
+For single-parameter suffixes this includes the following types:
+
++-------------------------------------------------------------+----------------------------------------------------------------+
+| Parameter type                                              | Accepted literals                                              |
++=============================================================+================================================================+
+| ``bool``                                                    | - :ref:`Boolean<booleans>` literals                            |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``uint8``, ..., ``uint256``, ``int8``, ..., ``int256``      | - :ref:`Rational literals<rational_literals>` (including zero) |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``address``                                                 | - :ref:`Address literals<address_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``bytes1``, ..., ``bytes32``                                | - Hexadecimal number literals (not for ``bytes20``)            |
+|                                                             | - :ref:`Hexadecimal string literals<hexadecimal_literals>`     |
```

Comment:
Apparently hexadecimal number literals do not have their own section. And the fact that hexadecimal string literals (can I even call them that?) do and it's called "hexadecimal literals" makes things confusing. It's hard to refer to just one of them in such a way that the reader won't mistake it for the other.